### PR TITLE
Feature/add tensorrt meson support

### PR DESCRIPTION
**Pull request by rrcarlosrodriguez**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -44,6 +44,37 @@ if get_option('enable-tflite')
...
+# Define library dependencies for TensorRT support
+if get_option('enable-tensorrt')
+
+  # CUDA versions supported
+  cuda_versions = [
+    '10.0',
+  ]
+
+  # Find CUDA version
+  cuda_ver = ''
```
Comment (jafet-chaves) [inline meson.build:56]: Why we are not looking for a specific cuda version?

Comment (rrcarlosrodriguez) [reply meson.build:56]: I implemented a mechanism to add supported CUDA versions, and Meson will look for any of them installed in the system

File: meson.build [hunk 2]
```diff
@@ -44,6 +44,37 @@ if get_option('enable-tflite')
...
+  foreach v : cuda_versions
+    if cuda_ver == ''
```
Comment (michaelgruner) [inline meson.build:58]: I'd break the loop instead of asking for this. This way we don't loop unnecessarily.

File: meson.build [hunk 3]
```diff
@@ -44,6 +44,37 @@ if get_option('enable-tflite')
...
+    endif
+  endforeach
+
+  if not cuda_dep.found() or not cudart_dep.found()
+    error ('Unable to find CUDA supported version')
+  endif
+
+  tensorrt = cpp.find_library('nvinfer', required: true)
+  tensorrt_dep = declare_dependency(dependencies: tensorrt)
+  cuda_inc_dir = cuda_dep.get_pkgconfig_variable('includedir')
```
Comment (michaelgruner) [inline meson.build:73]: This should be unnecessary. This is a symptom of a dependency not carried appropriately.

File: r2i/tensorrt/meson.build [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  'loader.h',
+  'model.h',
+  'parameters.h',
+  'prediction.h',
+]
+
+# Build library
+trt_lib = static_library('trt',
+  trt_sources,
+  include_directories : [configinc, cuda_inc_dir],
```
Comment (michaelgruner) [inline r2i/tensorrt/meson.build:25]: `cuda_inc_dir` should not be necessary if `lib_trt_dep` was configured appropriately.

**Pull request merged**